aeron-cluster/src/main/java/io/aeron/cluster/IngressAdapter.java

```diff
@@ -73,12 +72,12 @@ public Action onFragment(final DirectBuffer buffer, final int offset, final int
         messageHeaderDecoder.wrap(buffer, offset);
 
         final int schemaId = messageHeaderDecoder.schemaId();
+        final int templateId = messageHeaderDecoder.templateId();
         if (schemaId != MessageHeaderDecoder.SCHEMA_ID)
         {
-            throw new ClusterException("expected schemaId=" + MessageHeaderDecoder.SCHEMA_ID + ", actual=" + schemaId);
+            consensusModuleAgent.onUnknownMessage(schemaId, templateId, buffer, offset, length, header);
```

Comment:
Should return after this so the same template id from another schema does not get dispatched.